imap_processing/decom.py

```diff
@@ -1,5 +1,37 @@
+# Standard
+import logging
+
+from bitstring import ReadError
+
+# Installed
 from space_packet_parser import parser, xtcedef
 
+logging.basicConfig(level=logging.INFO)
+
+
+def safe_packet_generator(packet_parser, binary_data):
```

Comment:
I'm a little confused by this. I think this will actually keep going because you are catching the error and just logging it, but not re-raising.

More generally, I'm curious which behavior we actually want here? If we get a bad packet do we want to log and continue, possibly creating a file that is missing data and our log files are the only thing telling us that. Or do we want to raise and not produce a data file at all, so someone would have to come in and update the bad data.

Reply:
I'm trying to think of a better way to do it. The reason I did it that way was that my packets were all the same length (848 bits), but the very last packet had incomplete data. I assume this is when the simulator was turned off. So it had fewer bits than expected. I created safe_packet_generator to handle that last packet.

Do you think there would be real (not simulated) packets that are cutoff like that during normal processing? 

I am changing it to this, but not certain if that solves the problem. Thoughts?

    try:
        yield from packet_parser.generator(binary_data)
    except ReadError as e:
        if "Reading off the end of the data." in str(e):
            logging.warning(f"Acceptable read error encountered: {e}")
        else:
            logging.error(f"Unexpected read error: {e}")
            raise

Reply:
I didn't realize ReadError is actually from bitstring and not from space_packet_parser. I feel like space_packet_parser should actually not yield this packet if it isn't the right size to be able to read all of the data?

> Do you think there would be real (not simulated) packets that are cutoff like that during normal processing?
No, which is why I am somewhat worried that this could hide the fact that we got bad packets down during the mission by just logging and continuing on rather than raising and actually raising alarms on our end.
I see why you need to do this in this situation since you presumably got incomplete data though.

Some thoughts:
1) Did you verify with Ultra that the final packet is the wrong length? Maybe there is something else we need to be aware of and this caught something that went wrong on their end.
2) Can you strip that binary file down to be the right length and remove that final packet from the original file? Then you wouldn't need to do this dance.

Reply:
I've emailed them (CC'd you). We'll see what they say.